src/context/Search/useSearch.ts

```diff
@@ -26,38 +28,58 @@ type UseSearchQueryReturnType = {
 export const createSearchQuery = (
   partialQuery: Partial<SearchQuery>
 ): SearchQuery => {
-  const { term = "", keyStages = [] } = partialQuery;
-  return { term, keyStages };
+  const { term = "", keyStages = [], subjects = [] } = partialQuery;
+  return { term, keyStages, subjects };
 };
+
 const useSearchQuery = ({
   allKeyStages,
+  allSubjects,
 }: {
-  allKeyStages: KeyStage[];
+  allKeyStages?: KeyStage[];
+  allSubjects?: SearchPageData["subjects"];
 }): UseSearchQueryReturnType => {
   const {
-    query: { term = "", keyStages = "" },
+    query: { term = "", keyStages = "", subjects = "" },
     push,
   } = useRouter();
 
-  const isKeyStage = useCallback(
-    (slug: string) => {
-      return allKeyStages.some((keyStage) => keyStage.slug === slug);
+  const isFilterItem = useCallback(
+    <T extends { slug: string }>(slug: string, allFilterItems: T[]) => {
+      return allFilterItems.some((item) => item.slug === slug);
     },
-    [allKeyStages]
+    []
   );
 
-  const termString = term?.toString() || "";
-  const keyStagesArray = useMemo(
-    () => keyStages.toString().split(","),
-    [keyStages]
+  const getFilterForQuery = useCallback(
```

Comment:
Can this function and `isFilterItem` be a util defined outside of the hook function? It seems they're both pure standalone functions which could be tested independently

Reply:
Yeah good call, have moved these to the helper files and added tests